### PR TITLE
Initial pass at deletions support

### PR DESCRIPTION
**Pull request by malcolmholmes**
This is a first pass at adding deletions to Tanka (see #88 ). This code adds a label to a resource, then, if a flag is set, tracks down all resources with that label, and removes them if they aren't in the YAML.

The actual Delete() function isn't yet working/tested, but the logic seems to work for `tk diff`.

This is not considered complete - just proof that the idea works.

### CODE REVIEW

File: cmd/tk/workflow.go [hunk 1]
```diff
@@ -101,15 +118,26 @@ func diffCmd() *cobra.Command {
...
+			if err != nil {
+				log.Fatalln("Reconciling deletions:", err)
+			}
+			for _, deletion := range deletions {
+				fmt.Println("Requires deletion: ", deletion)
+			}
 		}
-		log.Println("No differences.")
+
+		os.Exit(16)
```
Comment (woodsaj) [inline cmd/tk/workflow.go]: This will result in the process always exiting with code 16, even if there are no diffs.

File: cmd/tk/workflow.go [hunk 2]
```diff
@@ -37,21 +37,35 @@ func applyCmd() *cobra.Command {
...
 		if err != nil {
 			log.Fatalln("Reconciling:", err)
 		}
 
+		if *willDelete {
+			deletions, err := kube.ReconcileDeletions(desired, kubernetes.ReconcileDeletionsOpts{Environment: environment})
+			if err != nil {
+				log.Fatalln("Reconciling deletions:", err)
+			}
+			err = kube.DeleteResources(deletions)
```
Comment (woodsaj) [inline cmd/tk/workflow.go]: you should pass `kubernetes.ApplyOpts{Force: *force, AutoApprove: *autoApprove}` to kube.DeleteResources

File: cmd/tk/workflow.go [hunk 3]
```diff
@@ -153,22 +181,35 @@ func showCmd() *cobra.Command {
...
 			log.Fatalln("Evaluating jsonnet:", err)
 		}
 
-		state, err := kube.Reconcile(raw, stringsToRegexps(vars.targets)...)
+		state, err := kube.Reconcile(raw, environment, stringsToRegexps(vars.targets)...)
 		if err != nil {
 			log.Fatalln("Reconciling:", err)
 		}
 
+		if *willDelete {
```
Comment (woodsaj) [inline cmd/tk/workflow.go]: As with apply, i feel that deletes should happen last.

File: pkg/kubernetes/kubernetes.go [hunk 4]
```diff
@@ -68,6 +71,9 @@ func (k *Kubernetes) Reconcile(raw map[string]interface{}, objectspecs ...*regex
 		if k != nil && !m.Has("metadata.namespace") {
 			m.Set("metadata.namespace", k.Spec.Namespace)
 		}
+		if k != nil && !m.Has("metadata.labels.tanka/origin") {
```
Comment (woodsaj) [inline pkg/kubernetes/kubernetes.go]: the label name should be defined as a constant.  Also, why are we using "origin", when everywhere else in this code we call it "environment"?

**Comment by malcolmholmes** [comment]
Replaced by #117

**Pull request closed without merging**